packages/evm/test/registry/guardians-registry-locking.js

```diff
@@ -0,0 +1,265 @@
+
+        const { amount, total } = await registry.getActivationLock(guardian, lockManager.address)
+        assertBn(amount, lockAmount.sub(unlockAmount.mul(bn(2))), 'locked amount does not match')
+        assertBn(total, lockAmount.sub(unlockAmount.mul(bn(2))), 'total locked amount does not match')
+      })
+
+      it('emits an event', async () => {
+        await registry.unlockActivation(guardian, lockManager.address, unlockAmount, false, { from })
+        const receipt = await registry.unlockActivation(guardian, lockManager.address, unlockAmount, false, { from })
+
+        assertAmountOfEvents(receipt, REGISTRY_EVENTS.GUARDIAN_ACTIVATION_LOCK_DECREASED)
+        assertEvent(receipt, REGISTRY_EVENTS.GUARDIAN_ACTIVATION_LOCK_DECREASED, { expectedArgs: { guardian, lockManager, amount: lockAmount.sub(unlockAmount.mul(bn(2))), total: lockAmount.sub(unlockAmount.mul(bn(2))) } })
+      })
+
+      it('allows to deactivate the unlocked amount', async () => {
+        await activateTokens(lockAmount)
+
+        await registry.unlockActivation(guardian, lockManager.address, unlockAmount, false, { from })
+
+        const receipt = await registry.deactivate(unlockAmount, { from: guardian })
+        assertEvent(receipt, REGISTRY_EVENTS.GUARDIAN_DEACTIVATION_REQUESTED, { decodeForAbi: GuardiansRegistry.abi, expectedArgs: { guardian, amount: unlockAmount } })
+      })
+    }
+
+    context('when the sender is not the lock manager', () => {
+      context('when the lock manager allows to unlock', () => {
+        beforeEach('mock can unlock', async () => {
+          await lockManager.mockCanUnlock(true)
+        })
+
+        context('when there was a locked amount', () => {
+          beforeEach('create lock', async () => {
+            await registry.changeLockManager(lockManager.address, true, { from: governor })
+            await registry.lockActivation(lockManager.address, lockAmount, { from: guardian })
+          })
+
+          context('when the sender is the guardian', () => {
+            const from = guardian
+
+            itUnlocksTheActivation(from)
+
+            it('can request a deactivation in the same call', async () => {
+              await activateTokens(lockAmount)
+
+              const receipt = await registry.unlockActivation(guardian, lockManager.address, unlockAmount, true, { from })
+
+              assertEvent(receipt, REGISTRY_EVENTS.GUARDIAN_DEACTIVATION_REQUESTED, { decodeForAbi: GuardiansRegistry.abi, expectedArgs: { guardian, amount: unlockAmount } })
+            })
+          })
+
+          context('when the sender is not the guardian', () => {
+            const from = someone
+
+            itUnlocksTheActivation(from)
+
+            it('can request a deactivation in the same call', async () => {
+              await activateTokens(lockAmount)
+
+              const receipt = await registry.unlockActivation(guardian, lockManager.address, unlockAmount, true, { from })
```

Comment:
Shouldn't this fail?

Reply:
what's the reasoning?

Reply:
Discussed offline: it's fine cause anyone can unlock activations